### PR TITLE
Make sweeper parallel per-topic again

### PR DESCRIPTION
**Pull request by saulius**
Adds global tasksToComplete list. After this list gets populated we wait for all tasks to complete. In case any of them fail - end the parent task with a rethrown exception.

@vidma @ljank @astrauka

### CODE REVIEW

**Comment by astrauka** [comment]
:+1:

File: camus-sweeper/src/main/java/com/linkedin/camus/sweeper/CamusSweeper.java [hunk 1]
```diff
@@ -222,8 +226,24 @@ public void run() throws Exception
...
+    while (remainingTaskCount > 0) {
+      completedTask = completionService.take();
+      remainingTaskCount--;
+
+      try {
+          completedTask.get();
+      } catch (ExecutionException e) {
+          // If a task completed with an exception shutdown all other tasks and exit
+          // re-raise the exception to make the parent job fail.
+          executorService.shutdown();
```
Comment (vidma) [inline camus-sweeper/src/main/java/com/linkedin/camus/sweeper/CamusSweeper.java]: `executorService.shutdown();`  [allows the tasks to continue running](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ExecutorService.html), maybe we need `.shutdownNow()` instead?

File: camus-sweeper/src/main/java/com/linkedin/camus/sweeper/CamusSweeper.java [hunk 2]
```diff
@@ -222,8 +226,24 @@ public void run() throws Exception
       {
         System.err.println("unable to process " + topicName + " skipping...");
         e.printStackTrace();
-        executorService.shutdown();
-        throw e;
```
Comment (vidma) [inline camus-sweeper/src/main/java/com/linkedin/camus/sweeper/CamusSweeper.java]: what if we caught and ignored a bad exception here?  what will be the `exit status`? `0=success`?

File: camus-sweeper/src/main/java/com/linkedin/camus/sweeper/CamusSweeper.java (hunk 1 above)
Comment (saulius) [inline camus-sweeper/src/main/java/com/linkedin/camus/sweeper/CamusSweeper.java]: `shutdownNow` [does that as well](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ExecutorService.html). However [a few lines down](https://github.com/vinted/camus/pull/26/files#diff-34802ddc212164b4c2a5af563768e90bR257) we'll be waiting for their termination or bailing out after timeout anyway.

File: camus-sweeper/src/main/java/com/linkedin/camus/sweeper/CamusSweeper.java (hunk 2 above)
Comment (saulius) [inline camus-sweeper/src/main/java/com/linkedin/camus/sweeper/CamusSweeper.java]: It's a bit hard to simulate "means unable to create a task/future". This is how Camus worked originally, I removed whole `try`/`catch` block and with a simple `Exception` it now exits with `1`.

File: camus-sweeper/src/main/java/com/linkedin/camus/sweeper/CamusSweeper.java (hunk 1 above)
Comment (vidma) [inline camus-sweeper/src/main/java/com/linkedin/camus/sweeper/CamusSweeper.java]: > However a few lines down we'll be waiting for their termination

but line below throws an Exception, so `awaitTermination` won't be reached. not sure if this is bad, maybe not much.

Comment (saulius) [inline camus-sweeper/src/main/java/com/linkedin/camus/sweeper/CamusSweeper.java]: It seems to work fine in both failure and success cases now.

**Comment by saulius** [comment]
Will make alternative, simplified variant of this.

**Pull request closed without merging**